main/power/tdisplays3.inc

```diff
@@ -1,14 +1,21 @@
-// tdisplays3 and devices with no power-management but it can measure voltage level of the battery and
-// detect charging status when a battery is connected through an analog input
+// T-Display S3 implementation without PMIC (ADC-based)
 //
+
 #include <driver/gpio.h>
 #include <esp_adc/adc_cali.h>
+#include <esp_adc/adc_cali_scheme.h>
 #include <esp_adc/adc_oneshot.h>
+#include <esp_sleep.h>
+
+#include "tusb.h"
 
 #define BATTERY_ADC_CHANNEL ADC_CHANNEL_3
+#define BATTERY_ADC_ATTEN ADC_ATTEN_DB_12
+#define BATTERY_EMA_ALPHA 0.3 // estimated moving average smoothing factor
```

Comment:
Please make this `0.3f` to avoid `double` math, and same where also defined below.